### PR TITLE
Create Jenkins_pipeline_npm.groovy

### PR DESCRIPTION
**Pull request by lukebroganws**
Jenkins NPM Pipeline example

### CODE REVIEW

File: CI-CD/Jenkins/Jenkins_pipeline_npm.groovy [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   }
+  
+  tools {
+    nodejs "nodejs-17.3.1"
+  }
+  
+  stages {
+    stage('Cloning Git') {
+      steps {
+        git 'https://github.com/lukebroganws/NodeGoat'
```
Comment (tabac-ws) [inline CI-CD/Jenkins/Jenkins_pipeline_npm.groovy:20]: Put a generic git link

File: CI-CD/Jenkins/Jenkins_pipeline_npm.groovy [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      }
+    }
+
+    stage('Download WS Script') {
+      steps {
+                script {
+                    if (fileExists('./wss-unified-agent.jar')) {
+                        echo "File already exists"
+                    } else {
+                            sh 'curl -LJO https://unified-agent.s3.amazonaws.com/wss-unified-agent.jar'
```
Comment (tabac-ws) [inline CI-CD/Jenkins/Jenkins_pipeline_npm.groovy:36]: we may need to add here the sha256 verification - let me know if you need help with it

**Review by tabac-ws — CHANGES_REQUESTED** [review]

File: CI-CD/Jenkins/Jenkins_pipeline_npm.groovy [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    stage('Download WS Script') {
+      steps {
+                sh '''if ! [ -f ./wss-unified-agent.jar ] 
+                then curl -fSL -R -JO https://unified-agent.s3.amazonaws.com/wss-unified-agent.jar 
+                if [[ "$(curl -sL https://unified-agent.s3.amazonaws.com/wss-unified-agent.jar.sha256)" != "$(sha256sum wss-unified-agent.jar)" ]] ;
+                then echo "Integrity Check Failed" 
+                exit -7 
+                fi 
+                fi'''
```
Comment (tabac-ws) [inline CI-CD/Jenkins/Jenkins_pipeline_npm.groovy:38]: Have you tested it?

Comment (lukebroganws) [reply CI-CD/Jenkins/Jenkins_pipeline_npm.groovy:38]: yep, all runs well!

Comment (tabac-ws) [reply CI-CD/Jenkins/Jenkins_pipeline_npm.groovy:38]: Hi @lukebroganws , can you change it per our conversation?

Comment (lukebroganws) [reply CI-CD/Jenkins/Jenkins_pipeline_npm.groovy:37]: All done @tabac-ws

**Review by tabac-ws — DISMISSED** [review]

File: CI-CD/Jenkins/Jenkins_pipeline_maven.groovy [hunk 4]
```diff
@@ -29,13 +29,18 @@ pipeline {
...
+                    echo "Downloading WhiteSource Unified Agent and Checking Integrity"
+                    sh 'curl -LJO https://unified-agent.s3.amazonaws.com/wss-unified-agent.jar'
+                    ua_jar_checksum=sh(returnStdout: true, script: "sha256sum 'wss-unified-agent.jar'")
+                    ua_integrity_file=sh(returnStdout: true, script: "curl -sL https://unified-agent.s3.amazonaws.com/wss-unified-agent.jar.sha256")
+                    if ("${ua_integrity_file}" == "${ua_jar_checksum}") {
+                        echo "Integrity Check Passed"
                     } else {
-                            sh 'curl -LJO https://unified-agent.s3.amazonaws.com/wss-unified-agent.jar'
+                        echo "Integrity Check Failed"
+                        exit 1
```
Comment (tabac-ws) [inline CI-CD/Jenkins/Jenkins_pipeline_maven.groovy:41]: Please remove exit 1 - we don't fail the build - just warn about it

File: CI-CD/Jenkins/Jenkins_pipeline_npm.groovy [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+              script {
+                    echo "Downloading WhiteSource Unified Agent and Checking Integrity"
+                    sh 'curl -LJO https://unified-agent.s3.amazonaws.com/wss-unified-agent.jar'
+                    ua_jar_checksum=sh(returnStdout: true, script: "sha256sum 'wss-unified-agent.jar'")
+                    ua_integrity_file=sh(returnStdout: true, script: "curl -sL https://unified-agent.s3.amazonaws.com/wss-unified-agent.jar.sha256")
+                    if ("${ua_integrity_file}" == "${ua_jar_checksum}") {
+                        echo "Integrity Check Passed"
+                    } else {
+                        echo "Integrity Check Failed"
+                        exit 1
```
Comment (tabac-ws) [inline CI-CD/Jenkins/Jenkins_pipeline_npm.groovy:41]: Please remove exit 1 - we don't fail the build - just warn about it

**Review by tabac-ws — CHANGES_REQUESTED** [review]

**Review by tabac-ws — APPROVED** [review]

**Pull request merged**